File: LeetCode/0645-set-mismatch/solution.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        std::unordered_set<int> seen;
        int count = 0;
        int dup = 0;
        for(int i=0;i<n;i++) {
            count+= nums[i];
            if(seen.count(nums[i])) {
                dup = nums[i];
            }
            else {
                seen.insert(nums[i]);
            }
        }
        
        return {dup, (((n + 1) * n) / 2) - (count - dup) };
    }
};

File: LeetCode/0645-set-mismatch/solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        Arrays.sort(nums);
        int dup = -1;
        int actualSum = nums[nums.length - 1];
        int correctSum = nums.length;
        for(int i=0;i<nums.length - 1;i++) {
            actualSum += nums[i];
            correctSum += i + 1;
            if(nums[i] == nums[i + 1]) {
                dup = nums[i];
            }
        }

        int[] res = { dup, correctSum - (actualSum - dup) };
        return res;
    }
}
